scripts/bosh-lite.sh

```diff
@@ -0,0 +1,367 @@
+
+  # Get SSH key pair from the state
+  SSH_DIR="${BOSH_LITE_CODEBASE_PATH}/.vagrant/.ssh"
+  mkdir -p "${SSH_DIR}"
+  export BOSH_LITE_PRIVATE_KEY="${SSH_DIR}/id_rsa"
+  if [ ! -f "${BOSH_LITE_PRIVATE_KEY}" ]; then
+    echo "SSH key is missing, regenerating..."
+    ssh-keygen -f "${BOSH_LITE_PRIVATE_KEY}" -P ""
+    chmod 600 "${BOSH_LITE_PRIVATE_KEY}"
+  fi
+
+  local aws_key_finger_print
+  aws_key_finger_print=$(get_aws_key_pair_finger_print)
+  local local_key_finger_print
+  local_key_finger_print=$(if [ -f "${BOSH_LITE_PRIVATE_KEY}.fingerprint" ]; then cat "${BOSH_LITE_PRIVATE_KEY}.fingerprint"; fi)
+
+  if [ "${aws_key_finger_print}" != "${local_key_finger_print}" ]; then
+    echo "Updating AWS Key ${BOSH_LITE_KEYPAIR}"
+    if [ "${aws_key_finger_print}" != "None" ]; then
+      aws ec2 delete-key-pair \
+        --region "${BOSH_LITE_REGION}" \
+        --key-name "${BOSH_LITE_KEYPAIR}" > /dev/null || true
+    fi
+    aws ec2 import-key-pair \
+      --region "${BOSH_LITE_REGION}" \
+      --key-name "${BOSH_LITE_KEYPAIR}" \
+      --public-key-material "$(cat "${BOSH_LITE_PRIVATE_KEY}.pub")" > /dev/null
+    get_aws_key_pair_finger_print > "${BOSH_LITE_PRIVATE_KEY}.fingerprint"
+  fi
+}
+
+delete_ssh_key_pair() {
+  aws ec2 delete-key-pair \
+    --region "${BOSH_LITE_REGION}" \
+    --key-name "${BOSH_LITE_KEYPAIR}" > /dev/null || true
+}
+
+init_subnet_and_security_group() {
+  # Query the subnet tagged as Name=bosh-lite-subnet-0
+  BOSH_LITE_SUBNET_ID=$(
+    aws ec2 describe-subnets \
+      --region "${BOSH_LITE_REGION}" \
+      --filters "Name=tag:Name,Values=${BOSH_LITE_SUBNET_TAG_NAME}" \
+      --query 'Subnets[0].SubnetId' \
+      --output text
+    )
+  export BOSH_LITE_SUBNET_ID
+  if ! aws ec2 describe-subnets --region "${BOSH_LITE_REGION}" --subnet-ids "${BOSH_LITE_SUBNET_ID}" > /dev/null; then
+    echo
+    echo "ERROR: Cannot find valid subnet with Tag '${BOSH_LITE_SUBNET_TAG_NAME}'. Did you create it first?"
+    exit 1
+  fi
+
+  # Query the security group tagged as Name=bosh-lite-office-access
+  BOSH_LITE_SECURITY_GROUP=$(
+    aws ec2 describe-security-groups \
+      --region "${BOSH_LITE_REGION}" \
+      --filters "Name=tag:Name,Values=${BOSH_LITE_SECURITY_GROUP_TAG_NAME}" \
+      --query 'SecurityGroups[0].GroupId' \
+      --output text
+    )
+  export BOSH_LITE_SECURITY_GROUP
+  if ! aws ec2 describe-security-groups --region "${BOSH_LITE_REGION}" --group-ids "${BOSH_LITE_SECURITY_GROUP}" > /dev/null; then
+    echo
+    echo "ERROR: Cannot find valid security group with Tag '${BOSH_LITE_SECURITY_GROUP_TAG_NAME}'. Did you create it first?"
+    exit 1
+  fi
+}
+
+run_vagrant() {
+  (
+    cd "${BOSH_LITE_CODEBASE_PATH}"
+    vagrant "$@"
+  )
+}
+
+configure_auto_halt() {
+  if [ "${BOSH_LITE_AUTO_HALT}" == "true" ]; then
+    # m h dom mon dow user  command
+    echo "0 19 * * * root /sbin/halt -p" | run_vagrant ssh -- sudo tee /etc/cron.d/auto_halt > /dev/null
```

Comment:
@keymon showed me IRL that the cat redirection would not happen as root so we would have to wrap the command and the redirection in a subshell for example. In the end it would be uglier.